src/citrine/ara/variables.py

```diff
@@ -101,3 +110,232 @@ def __init__(self, *,
         self.name = name
         self.headers = headers
         self.template = template
+
+
+class AttributeByTemplateAfterProcessTemplate(Serializable['AttributeByTemplateAfterProcessTemplate'], Variable):
+    """Attribute of an object that is marked by an attribute template and derives from a process
+    marked by a locally unique object template.
+
+    TODO: define "locally-unique"
+
+    Parameters
+    ---------
+    name: str
+        a short human-readable name to use when referencing the variable
+    headers: list[str]
+        sequence of column headers
+    attributeTemplate: LinkByUID
+        attribute template that identifies the attribute to assign to the variable
+    processTemplate: LinkByUid
+        process template that identifies the originating process
+    """
+
+    name = properties.String('name')
+    headers = properties.List(properties.String, 'headers')
+    attributeTemplate = properties.Object(LinkByUID, 'attributeTemplate')
+    processTemplate = properties.Object(LinkByUID, 'processTemplate')
+    typ = properties.String('type', default="attribute_after_process", deserializable=False)
+
+    def _attrs(self) -> List[str]:
+        return ["name", "headers", "attributeTemplate", "processTemplate", "typ"]
+
+    def __init__(self, *,
+                 name: str,
+                 headers: List[str],
+                 attributeTemplate: LinkByUID,
+                 processTemplate: LinkByUID):
+        self.name = name
+        self.headers = headers
+        self.attributeTemplate = attributeTemplate
+        self.processTemplate = processTemplate
+
+
+class AttributeByTemplateAndObjectTemplate(Serializable['AttributeByTemplateAndObjectTemplate'], Variable):
+    """Attribute marked by an attribute template and an object template.
+
+    TODO: clarify documentation of use case here.
```

Comment:
An example would be if the same property is measured with two different measurement techniques.  The property would have the same attribute template, but the measurements would have different measurement templates, allowing them to be distinguished.